lib/ansible/plugins/action/uri.py

```diff
@@ -58,7 +60,8 @@ def run(self, tmp=None, task_vars=None):
                     raise AnsibleActionFail(
                         'body must be mapping, cannot be type %s' % body.__class__.__name__
                     )
-                for field, value in body.items():
+                new_body = deepcopy(body)
+                for field, value in new_body.items():
```

Comment:
I can't comment on this line, since it's too far out of the diff, but `kwargs['body'] = body` should be `kwargs['body'] = new_body` or otherwise we're not using the modified dict anywhere.